src/docs/codelabs/layout-basics_migrated.md

```diff
@@ -0,0 +1,3340 @@
+---
+title: "Basic Flutter layout concepts"
+description: "A codelab that teaches basic Flutter layout concepts through DartPad examples and exercises."
+toc: true
+js:
+- defer: true
+  url: https://dartpad.dev/inject_embed.dart.js
+---
+Welcome to the Flutter layout codelab,
+where you learn how to build a Flutter UI without
+downloading and installing Flutter or Dart!
+
+{{site.alert.important}}
+  This codelab covers basic Flutter layout concepts using an
+  experimental code editor called DartPad.
+  DartPad hasn't been fully tested on all browsers.
+  If you experience any difficulties while using DartPad
+  on a specific browser, please create a [DartPad issue][]
+  and specify which browser you're using in the issue title.
+{{site.alert.end}}
+
+Flutter is different from other frameworks because its UI
+is built in code, not (for example) in an XML file or similar.
+Widgets are the basic building blocks of a Flutter UI.
+As you progress through this codelab,
+you'll learn that almost everything in Flutter is a widget.
+A widget is an immutable object that describes a specific part of a UI.
+You'll also learn that Flutter widgets are composable, meaning,
+that you can combine existing widgets to make more sophisticated widgets.
+At the end of this codelab,
+you'll get to apply what you've learned
+into building a Flutter UI that displays a business card.
+
+**Estimated time to complete this codelab: 45-60 minutes.**
+
+## Row and Column classes
+`Row` and `Column` are classes that contain and lay out widgets.
+Widgets inside of a `Row` or `Column` are called *children*,
+and `Row` and `Column` are referred to as *parents*.
+`Row` lays out its widgets horizontally,
+and `Column` lays out its widgets vertically.
+
+#### Example: Creating a Column
+{:.no_toc}
+{{site.alert.secondary}}
+{:.no_toc}
+  The following example displays the differences between
+  a `Row` and `Column`.
+
+  **1.** Click the **Run** button.
+
+  **2.** In the code, change the `Row` to a `Column`, and run again.
+{{site.alert.end}}
+
+```run-dartpad:theme-dark:mode-flutter:run-true:width-100%:height-400px:split-60:null_safety-true
```

Comment:
`run-true` should be taken out here and in the others. Right now it's causing every DartPad embed on the page to execute at the same time, which is a CPU hit for both the browser and our compilation backend. 😄

Reply:
Ouch! I see, removed now.